### PR TITLE
fix(grid): fix banded columns layout in virtual table

### LINKED ISSUE #1389: Banded columns and virtual table bug
I use VirtualTable and BandedColumns together, column header cells behave strangely when horizontal scrolling. I think colspan is calculated incorrectly for banded columns. When the first column from group is removed from DOM, its banded header cell it is also removed

- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [x] React Grid
- [ ] React Chart
- [ ] Vue Grid

### PR DESCRIPTION
**Pull request by ushkal**
fixes #1389

### CODE REVIEW

File: packages/dx-grid-core/src/plugins/table-band-header/helpers.js [hunk 1]
```diff
@@ -31,87 +32,158 @@ export const getColumnMeta = (
...
+  return (prevColumnBand.title === currentColumnBand.title
+    && !currentTableColumn.rightColumnIndex
+    && (!isPrevColumnFixed || (isPrevColumnFixed && isCurrentColumnFixed)));
+};
+
+const columnRenderedInOtherRow = ({ currentColumnBand, currentRowLevel }) => (
+  currentColumnBand.level < currentRowLevel
+);
+
+const columnBelongsToCurrentRow = ({ currentColumnBand, currentRowLevel }) => (
```
Comment (MaximKudriavtsev) [inline packages/dx-grid-core/src/plugins/table-band-header/helpers.js]: Can we rename this functions similarly? I mean `columnRenderedInOtherRow` and (`columnBelongsToCurrentRow` => `columnRenderedInCurrentRow`)

Comment (ushkal) [reply packages/dx-grid-core/src/plugins/table-band-header/helpers.js]: these functions contain slightly different logic: 
`columnRenderedInOtherRow` means that a current header cell has _already_ been rendered in some upper row (and we replace it with empty cell), 
`columnBelongsToCurrentRow` returns true for a cell that has not been yet rendered.
We can consider names like `shouldRenderColumnInCurrentRow`, `isColumnInCurrentRow` or something similar

Comment (MaximKudriavtsev) [reply packages/dx-grid-core/src/plugins/table-band-header/helpers.js]: I prefer `isColumnInCurrentRow` or `isCurrentRowColumn`

Comment (SergeyAlexeev) [reply packages/dx-grid-core/src/plugins/table-band-header/helpers.js]: `shouldRenderColumnInCurrentRow` and `isColumnInCurrentRow`

+1

**Pull request closed without merging**